tools/flaky_test_monitor/common/testdata/test_data.go

```diff
@@ -0,0 +1,888 @@
+		level1TestResult7Rows = append(level1TestResult7Rows, getPassedTestPackage("TestEncodableStakingPrivKeyNil", "github.com/onflow/flow-go/model/encodable"))
+		level1TestResult7Rows = append(level1TestResult7Rows, getPassedTestPackage("TestEncodableStakingPubKey", "github.com/onflow/flow-go/model/encodable"))
+		level1TestResult7Rows = append(level1TestResult7Rows, getPassedTestPackage("TestEncodableStakingPubKeyNil", "github.com/onflow/flow-go/model/encodable"))
+		level1TestResult7Rows = append(level1TestResult7Rows, getPassedTestPackage("TestIsHexString", "github.com/onflow/flow-go/model/encodable"))
+		level1TestResult7Rows = append(level1TestResult7Rows, getNilTest_TestEncodableRandomBeaconPrivKeyMsgPack())
+	}
+
+	var level1TestResult8Rows []common.Level1TestResultRow
+	level1TestResult8Rows = append(level1TestResult8Rows, getNilTest_TestEncodableRandomBeaconPrivKeyMsgPack())
+
+	var level1TestResult9Rows []common.Level1TestResultRow
+	for i := 0; i < 4; i++ {
+		level1TestResult9Rows = append(level1TestResult9Rows, getNilTest_TestEncodableRandomBeaconPrivKeyMsgPack())
+	}
+	level1TestResult9Rows = append(level1TestResult9Rows, getPassedTestPackage("TestEncodableRandomBeaconPrivKeyMsgPack", "github.com/onflow/flow-go/model/encodable"))
+
+	var level1TestResult10Rows []common.Level1TestResultRow
+	for i := 0; i < 5; i++ {
+		level1TestResult10Rows = append(level1TestResult10Rows, getNilTest_TestEncodableRandomBeaconPrivKeyMsgPack())
+	}
+
+	level1Summary1 := common.Level1Summary{
+		Rows: level1TestResult1Rows,
+	}
+
+	level1Summary2 := common.Level1Summary{
+		Rows: level1TestResult2Rows,
+	}
+
+	level1Summary3 := common.Level1Summary{
+		Rows: level1TestResult3Rows,
+	}
+
+	level1Summary4 := common.Level1Summary{
+		Rows: level1TestResult4Rows,
+	}
+
+	level1Summary5 := common.Level1Summary{
+		Rows: level1TestResult5Rows,
+	}
+
+	level1Summary6 := common.Level1Summary{
+		Rows: level1TestResult6Rows,
+	}
+
+	level1Summary7 := common.Level1Summary{
+		Rows: level1TestResult7Rows,
+	}
+
+	level1Summary8 := common.Level1Summary{
+		Rows: level1TestResult8Rows,
+	}
+
+	level1Summary9 := common.Level1Summary{
+		Rows: level1TestResult9Rows,
+	}
+
+	level1Summary10 := common.Level1Summary{
+		Rows: level1TestResult10Rows,
+	}
+
+	var level1Summaries []common.Level1Summary
+	level1Summaries = append(level1Summaries, level1Summary1)
+	level1Summaries = append(level1Summaries, level1Summary2)
+	level1Summaries = append(level1Summaries, level1Summary3)
+	level1Summaries = append(level1Summaries, level1Summary4)
+	level1Summaries = append(level1Summaries, level1Summary5)
+	level1Summaries = append(level1Summaries, level1Summary6)
+	level1Summaries = append(level1Summaries, level1Summary7)
+	level1Summaries = append(level1Summaries, level1Summary8)
+	level1Summaries = append(level1Summaries, level1Summary9)
+	level1Summaries = append(level1Summaries, level1Summary10)
+	return level1Summaries
+}
```

Comment:
Could you add some comments here? It's not super clear what this function is doing. In particular, what is the meaning of the loop iteration counts?

Reply:
I added comments and more explanation for all the test data functions so it's more clear how they're used by the level 1 and 2 unit tests. This function has to create a collection of level 1 summaries that are used as inputs to the level 2 parser. The loop iterations model how many times `go test` was run (e.g. `go test -count=10`) to generate level 1 summaries. We then take a collection of all the level 1 summaries to generate a level 2 summary.

https://github.com/onflow/flow-go/pull/2102/commits/b8fcd03c948869f46a4a3566d1b3f6ee22388fba

Reply:
I added more comments on the different loop counters so it's more clear on why we use different loop counters to construct level 1 expected data.

https://github.com/onflow/flow-go/pull/2102/commits/0958e2012babadc413e4227a551d9d7b8ebbbbb5